feat(UgcCommonModelTrait): ✨ add geometry synchronization to UGC properties OC:6698

carlacupani opened the pull request:
- Introduced a new feature to synchronize geometry data into properties.position within UGC models during the saved event.
- Implemented a method to extract latitude and longitude from the UGC model's geometry using GeometryComputationService.
- Updated properties in the database only if the geographical coordinates have changed.
- Added error handling to log any exceptions encountered during the synchronization process.
- Modified the condition to check for changes in the validated field using wasChanged instead of isDirty for accuracy.

carlacupani committed:
feat(UgcCommonModelTrait): ✨ add geometry synchronization to UGC properties OC:6698

- Introduced a new feature to synchronize geometry data into properties.position within UGC models during the saved event.
- Implemented a method to extract latitude and longitude from the UGC model's geometry using GeometryComputationService.
- Updated properties in the database only if the geographical coordinates have changed.
- Added error handling to log any exceptions encountered during the synchronization process.
- Modified the condition to check for changes in the validated field using wasChanged instead of isDirty for accuracy.

peppedeka commented on app/Traits/UgcCommonModelTrait.php:
```diff
@@ -69,15 +70,109 @@ protected static function bootUgcCommonModelTrait()
+                            ];
+
+                            // Aggiorna properties senza triggerare altri observer per evitare loop
+                            DB::table($table)
```
perche non hai usato saveQuietly o updateQuitetly?

carlacupani committed:
refactor(geometry): ♻️ streamline geometry conversion and update logic OC:6698

- Replace direct GeoJSON to geometry conversion with `GeometryComputationService` in `GeometryService`.
- Simplify `UgcCommonModelTrait` by removing redundant database reads for properties.
- Utilize `updateQuietly` to prevent event loops during property updates.
- Maintain functionality while enhancing code readability and maintainability.

carlacupani committed:
refactor(geometry-service): ♻️ use geojsonToGeometry directly within the service

- Replace the use of `GeometryComputationService::make()` with direct invocation of `geojsonToGeometry` method within the `GeometryService`.
- This change simplifies the code by removing unnecessary service instantiation, leveraging the existing method in the same service for converting GeoJSON to geometry.

carlacupani committed:
fix(ugc): 🐛 use isDirty instead of wasChanged for validation check OC:6698

- Replaced `wasChanged` with `isDirty` to accurately detect if the 'validated' field has been modified before creating an EC Poi from validated UGC.
- Ensures that the logic triggers correctly upon changes to the 'validated' status.
